app/uk/gov/hmrc/plasticpackagingtax/registration/connectors/GrsConnector.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.registration.connectors
+
+import com.kenshoo.play.metrics.Metrics
+import play.api.http.Status.CREATED
+import play.api.libs.json.{Reads, Writes}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse, InternalServerException}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+abstract class GrsConnector[
+  GrsCreateJourneyPayload,
+  GrsResponse
+] @Inject() (
+  httpClient: HttpClient,
+  metrics: Metrics,
+  grsUrl: String,
+  createJourneyTimerTag: String,
+  getJourneyDetailsTimerTag: String
+)
+(implicit ec: ExecutionContext) {
+  type RedirectUrl  = String
```

Comment:
Type aliases like this are generally not a great idea. They don't create newtype semantics (i.e. in this case, there is no difference between `String` and `RedirectUrl` to the compiler). And although aliases _can_ be descriptive and useful, it's generally unhelpful for simple types like `String`.

Reply:
I feel there is some value here as it makes it clear to the reader that the function returns a "redirect URL".

Reply:
Consider creating a case class with a single string value e.g.
`case class RedirectUrl(url: String) extends AnyVal`
You get the clarity without the type alias.